### PR TITLE
Full upgrade to RxJava 2

### PR DESCRIPTION
**Pull request by tushar-acharya**

### CODE REVIEW

**Comment by pwittchen** [comment]
Hi @tushar-acharya. Thanks for creating this PR. I'll review it as soon as I can.
For sure, I want to keep compatibility with RxJava1.x, so I'll create a separate branch `RxJava2.x` for that changes and RxJava2.x version is going to be released as a separate artifact. When the branch will be created, I'll notify you, so you can change target branch from `master` to `RxJava2.x`

File: app/src/main/java/com/github/pwittchen/reactivenetwork/app/MainActivity.java [hunk 1]
```diff
@@ -47,9 +47,8 @@
     networkConnectivitySubscription =
         ReactiveNetwork.observeNetworkConnectivity(getApplicationContext())
             .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .subscribe(new Action1<Connectivity>() {
-              @Override public void call(final Connectivity connectivity) {
+            .observeOn(AndroidSchedulers.mainThread()).subscribe(new Consumer<Connectivity>() {
```
Comment (pwittchen) [inline app/src/main/java/com/github/pwittchen/reactivenetwork/app/MainActivity.java]: Please, put subscribe in the new line.

File: app/src/main/java/com/github/pwittchen/reactivenetwork/app/MainActivity.java [hunk 2]
```diff
@@ -59,9 +58,8 @@
 
     internetConnectivitySubscription = ReactiveNetwork.observeInternetConnectivity()
         .subscribeOn(Schedulers.io())
-        .observeOn(AndroidSchedulers.mainThread())
-        .subscribe(new Action1<Boolean>() {
-          @Override public void call(Boolean isConnectedToInternet) {
+        .observeOn(AndroidSchedulers.mainThread()).subscribe(new Consumer<Boolean>() {
```
Comment (pwittchen) [inline app/src/main/java/com/github/pwittchen/reactivenetwork/app/MainActivity.java]: Please, put subscribe in the new line.

File: build.gradle [hunk 3]
```diff
@@ -4,14 +4,14 @@ ext {
...
-  gradleVersion = '2.2.1'
+  gradleVersion = '2.3.0'
   kotlinVersion = '1.1.0'
 }
 
-ext.deps = [rxjava            : 'io.reactivex:rxjava:1.2.7',
-            rxandroid         : 'io.reactivex:rxandroid:1.2.1',
-            supportannotations: 'com.android.support:support-annotations:25.2.0',
-            appcompatv7       : 'com.android.support:appcompat-v7:25.2.0',
+ext.deps = [rxjava            : 'io.reactivex.rxjava2:rxjava:2.0.7',
```
Comment (pwittchen) [inline build.gradle]: Please, change `rxjava` variable to `rxjava2` variable.

File: build.gradle [hunk 4]
```diff
@@ -4,14 +4,14 @@ ext {
...
+            rxandroid         : 'io.reactivex.rxjava2:rxandroid:2.0.1',
```
Comment (pwittchen) [inline build.gradle]: Please change `rxandroid` variable to `rxandroid2` variable.

File: library/src/main/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/SocketInternetObservingStrategy.java [hunk 5]
```diff
@@ -56,8 +57,8 @@
     Preconditions.checkNotNull(errorHandler, "errorHandler is null");
 
     return Observable.interval(initialIntervalInMs, intervalInMs, TimeUnit.MILLISECONDS,
-        Schedulers.io()).map(new Func1<Long, Boolean>() {
-      @Override public Boolean call(Long tick) {
+        Schedulers.io()).map(new Function<Long, Boolean>() {
+      @Override public Boolean apply(@NonNull Long aLong) throws Exception {
```
Comment (pwittchen) [inline library/src/main/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/SocketInternetObservingStrategy.java]: Please change `aLong` variable name to `tick` as it was previously.

File: library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java [hunk 6]
```diff
@@ -44,8 +45,9 @@
     final IntentFilter filter = new IntentFilter();
     filter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
 
-    return Observable.create(new Observable.OnSubscribe<Connectivity>() {
-      @Override public void call(final Subscriber<? super Connectivity> subscriber) {
+    return Observable.create(new ObservableOnSubscribe<Connectivity>() {
+      @Override public void subscribe(final ObservableEmitter<Connectivity> subscriber)
```
Comment (pwittchen) [inline library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java]: Please change `subscriber` name to `emmiter`. Type was changed, so in this case, we should change variable name as well.

File: library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java [hunk 7]
```diff
@@ -75,17 +77,21 @@ private void tryToUnregisterReceiver(final Context context, final BroadcastRecei
...
-          inner.schedule(new Action0() {
-            @Override public void call() {
-              unsubscribe.call();
-              inner.unsubscribe();
+          inner.schedule(new Runnable() {
+            @Override public void run() {
+              try {
+                unsubscribe.run();
+              } catch (Exception e) {
+                e.printStackTrace();
```
Comment (pwittchen) [inline library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java]: Please, do not print the whole StackTrace. Call `onError(message, exception)` method with an appropriate message.

File: library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java [hunk 8]
```diff
@@ -89,7 +89,7 @@ private Disposable unsubscribeInUiThread(final Action unsubscribe) {
               try {
                 unsubscribe.run();
               } catch (Exception e) {
-                e.printStackTrace();
+                onError(LOG_TAG, e);
```
Comment (pwittchen) [inline library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java]: There should not be a `LOG_TAG`! This method has the following signature `onError(final String message, final Exception exception)`. Please put there a descriptive message. E.g. `Could not dispose subscription in UI Thread`.

**Comment by pwittchen** [comment]
I've performed code review. Please, take a look at my comments & perform updates.

**Comment by tushar-acharya** [comment]
Thanks for the review. I'll fix it

**Comment by pwittchen** [comment]
I've created [RxJava2.x](https://github.com/pwittchen/ReactiveNetwork/tree/RxJava2.x) branch from the master branch. **Please change target branch** in this PR from `master` branch to `RxJava2.x` branch.

**Comment by tushar-acharya** [comment]
@pwittchen sure will do that

**Comment by tushar-acharya** [comment]
@pwittchen Awesome 😄 Thanks

**Comment by tushar-acharya** [comment]
@pwittchen Let me know if you need any help in testing any specific scenarios. We are currently using this as an extra module. It would really help if we can release this as a jcenter dependency :) No rush though

**Comment by pwittchen** [comment]
I'm gonna take care of that this week.

**Comment by pwittchen** [comment]
Changes are merged. I performed pre-release preparations. I'm gonna release new version soon (probably on Monday or on Tuesday). I just need to go through my checklist to ensure that everything is fine.

**Pull request merged**